### ISSUE TITLE
Memory leak for RAM MEM

### ISSUE THREAD

**Issue by cookielee77** [ruotianluo/ImageCaptioning.pytorch#84]
It seems there is a memory leak problem on RAM (NOT on GPU) when training the model. During the training, the free memory on RAM keeps decreasing, especially when using att_feats (I used **free -m** to monitor the RAM usage). I debugged a little bit and found this issue is caused by the data from Dataloader, and didn't free the data at somehow. 

![screenshot from 2019-01-18 17 06 19](https://user-images.githubusercontent.com/13234411/51419976-7c14af80-1b43-11e9-966d-d504aa2e9581.png)

Not sure the exact codes cause this. Did you notice this situation before? Thanks for the help!

**Comment by mememimis** [comment]
That's true. I'm still trying to find out what causes this problem. Did you find a solution?

**Comment by cookielee77** [comment]
Nope. I didn't debug too many details. I'm not familiar with the author's dataloader type. Maybe the author could fix this at some time.

**Comment by zhangqipeng99** [comment]
I have the same problem. please tell me,Have you solved it?

**Comment by prajwalkr** [comment]
In file dataloader.py line 241, change num_workers to a small enough number, and after that, I did not run out of memory.
